File: Java/src/main/java/account/DebitCard.java
package account;

public class DebitCard extends PaymentAccount {

    private int balance;

    public DebitCard(String accountHolder, int accountNumber, int balance) {
        super(accountHolder, accountNumber);
        this.balance = balance;
    }

    public boolean pay(int amount) {
        if (this.balance >= amount) {
            this.updateBalance(amount);
            return true;
        }
        return false;
    }

    private void updateBalance(int amount) {
        this.balance -= amount;
    }
}

File: TypeScript/src/accounts/debitcard.ts
import { PaymentAccount } from "./paymentaccount";

class DebitCard extends PaymentAccount {
	
    private balance: number;

    constructor(accountHolder: string, accountNumber: number, balance: number) {
        super(accountHolder, accountNumber);
        this.balance = balance;
    }

    public pay(amount: number): boolean {
        if (amount > this.balance) {
            return false;
        }
        this.balance -= amount;
        return true;
    }

    public getbalance(): number {
        return this.balance;
    }
}